qemu/tests/hv_vapic_test.py

```diff
@@ -0,0 +1,130 @@
+import logging
+import re
+import os
+import time
+
+from virttest import error_context
+from virttest import utils_disk
+from virttest import env_process
+from virttest import data_dir
+from avocado.utils import process
+from virttest.qemu_storage import QemuImg
+from provider.storage_benchmark import generate_instance
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test the hv_vapic flag improvement
+    1) Create tmpfs data disk in host
+    2) Mount&format the disk in guest, then prepare the fio test
+       environment
+    3) Boot the guest with all hv flags
+    4) Run fio test, record the result's bw value
+    5) Shutdown and boot the guest again without hv_vapic flag
+    6) Run fio test again, record the result's bw value
+    7) Calculate the improvement value of the 2 fio tests,
+       then check if it is obvious enough
+
+    param test: the test object
+    param params: the test params
+    param env: the test env object
+    """
+
+    def _create_tmpfs_data_disk():
+        """
+        Create a tmpfs data disk
+        """
+        logging.info("Create tmpfs data disk")
+        disk_name_key = params["disk_name_key"]
+        tmp_dir = data_dir.get_tmp_dir()
+        tmpfs_dir = os.path.join(tmp_dir, "tmpfs")
+        process.system("mkdir -p " + tmpfs_dir, shell=True)
+        params[disk_name_key] = os.path.join(tmpfs_dir, "data")
+        tmpfs_image_name = params["tmpfs_image_name"]
+        img_param = params.object_params(tmpfs_image_name)
+        img = QemuImg(img_param, data_dir.get_data_dir(), tmpfs_image_name)
+        img.create(img_param)
+
+    def _format_tmpfs_disk():
+        """
+        Format the new tmpfs disk in guest
+
+        return: the formatted drive letter of the disk
+        """
+        logging.info("Boot the guest to setup tmpfs disk")
+        vm, session = _boot_guest_with_cpu_flag(cpu_model_flags)
+        logging.info("Format tmpfs disk")
+        disk_size = params["image_size_" + params["tmpfs_image_name"]]
+        disk_id = utils_disk.get_windows_disks_index(session, disk_size)[0]
+        drive_letter = utils_disk.configure_empty_windows_disk(
+                    session, disk_id, disk_size)[0]
+        vm.graceful_shutdown(timeout=timeout)
+        return drive_letter
+
+    def _boot_guest_with_cpu_flag(hv_flag):
+        """
+        Boot the guest, with param cpu_model_flags set to hv_flag
+
+        param hv_flag: the hv flags to set to cpu
+
+        return: the booted vm and a loggined session
+        """
+        params["cpu_model_flags"] = hv_flag
+        params["start_vm"] = "yes"
+        vm_name = params["main_vm"]
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(vm_name)
```

Comment:
Better to add `vm.verify_alive()` after `vm = env.get_vm(vm_name)` to verify some info:
```
        self.verify_disk_image_bootable()
        self.verify_userspace_crash()
        self.verify_kernel_crash()
        self.verify_illegal_instruction()
        self.verify_kvm_internal_error()
```